src/kemal/middleware/filters.cr

```diff
@@ -74,7 +74,10 @@ module Kemal::Middleware
     property block : HTTP::Server::Context -> String
 
     def initialize(&block : HTTP::Server::Context -> _)
-      @block = ->(context : HTTP::Server::Context) { block.call(context).to_s }
+      @block = ->(context : HTTP::Server::Context) do 
+        block.call(context).to_s.to_json if context.request.headers["Content-Type"]? == "application/json"
```

Comment:
Use `if/else`, and just `to_json`, not `to_s.to_json`. I'd also like to stream the result to the context's response, but with the current way it doesn't seem to be possible. I might add a PR that changes all of this later, if you don't mind.